pkg/workflow/workflow.go

```diff
@@ -30,16 +31,9 @@ var (
 	DisableRecorder = false
 )
 
-const (
-	// MessageTerminatedFailedAfterRetries is the message of failed after retries
-	MessageTerminatedFailedAfterRetries = "The workflow terminates automatically because the failed times of steps have reached the limit"
-	// MessageSuspendFailedAfterRetries is the message of failed after retries
-	MessageSuspendFailedAfterRetries = "The workflow suspends automatically because the failed times of steps have reached the limit"
-)
-
 // IsFailedAfterRetry check if application is hang due to FailedAfterRetry
 func IsFailedAfterRetry(app *oamcore.Application) bool {
-	return app.Status.Workflow != nil && (app.Status.Workflow.Message == MessageTerminatedFailedAfterRetries || app.Status.Workflow.Message == MessageSuspendFailedAfterRetries)
```

Comment:
why terminatedFailed not needed?

Reply:
If the workflow is failed, the status is now `failed` instead of `termianted`.